### PR TITLE
cleanups and quesions for preview generation

### PR DESCRIPTION
**Pull request by FZambia**
Some things to consider for #44891

### CODE REVIEW

File: pkg/services/thumbs/crawler.go [hunk 1]
```diff
@@ -62,35 +65,36 @@ func (r *simpleCrawler) next() *models.DashboardWithStaleThumbnail {
...
+		r.logger.Info("Crawler already running")
 		return r.Status()
 	}
 
-	r.queueMutex.Lock()
-	defer r.queueMutex.Unlock()
-
 	now := time.Now()
 
+	// TODO: how theming work with stale thumbnails? I tried to start crawler with light theme but no stale dashboards found.
```
Comment (ArturWierzbicki) [inline pkg/services/thumbs/crawler.go:94]: resolved it in https://github.com/grafana/grafana/pull/44891/files

File: pkg/services/thumbs/crawler.go [hunk 2]
```diff
@@ -139,16 +144,27 @@ func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme mode
...
-
-	// create a pool of workers
-	for i := 0; i < r.threadCount; i++ {
-		go r.walk()
+	r.logger.Info("Starting dashboard crawler", "dashboardsToCrawl", len(items))
 
-		// wait 1/2 second before starting a new thread
-		time.Sleep(500 * time.Millisecond)
-	}
+	go func() {
```
Comment (ArturWierzbicki) [inline pkg/services/thumbs/crawler.go:149]: done in https://github.com/grafana/grafana/pull/44891/files

File: pkg/services/thumbs/crawler.go [hunk 3]
```diff
@@ -139,16 +144,27 @@ func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme mode
...
+			// wait 1/2 second before starting a new thread
+			// TODO: why we are sleeping here?
+			time.Sleep(500 * time.Millisecond)
+		}
+		wg.Wait()
+		r.walkFinished()
+		r.broadcastStatus()
+	}()
 
+	// TODO: what we broadcasting here?
```
Comment (ArturWierzbicki) [inline pkg/services/thumbs/crawler.go:167]: initial status, ie. running = started

File: pkg/services/thumbs/crawler.go [hunk 4]
```diff
@@ -232,23 +248,24 @@ func (r *simpleCrawler) walk() {
...
+			r.logger.Warn("Error getting image... no response", "dashboardUID", item.Uid, "url", url)
 			r.newErrorResult()
 		} else if strings.Contains(res.FilePath, "public/img") {
-			tlog.Warn("error getting image... internal result", "dashboardUID", item.Uid, "url", url, "img", res.FilePath)
+			r.logger.Warn("Error getting image... internal result", "dashboardUID", item.Uid, "url", url, "img", res.FilePath)
 			r.newErrorResult()
 		} else {
 			func() {
 				defer func() {
+					// TODO: should we remove this file for previous if/else branches?
```
Comment (ArturWierzbicki) [inline pkg/services/thumbs/crawler.go:265]: good catch, added removal in the 3rd branch.

Comment (ArturWierzbicki) [reply pkg/services/thumbs/crawler.go:265]: oh wait no, these are static images so we should not remove them

Comment (ArturWierzbicki) [reply pkg/services/thumbs/crawler.go:265]: added a comment instead

File: pkg/services/thumbs/crawler.go (hunk 3 above)
Comment (FZambia) [reply pkg/services/thumbs/crawler.go:167]: But there is one more broadcast above, before starting goroutines

File: pkg/services/thumbs/crawler.go [hunk 5]
```diff
@@ -115,12 +119,13 @@ func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme mode
 		Theme:           theme,
 		ConcurrentLimit: 10,
 	}
+	// TODO: Should we create session if no items found by findDashboardsWithStaleThumbnails?
```
Comment (FZambia) [inline pkg/services/thumbs/crawler.go:122]: ?

File: pkg/services/thumbs/crawler.go [hunk 6]
```diff
@@ -139,16 +144,27 @@ func (r *simpleCrawler) Start(c *models.ReqContext, mode CrawlerMode, theme mode
...
+		var wg sync.WaitGroup
+		wg.Add(r.threadCount)
+		// create a pool of workers
+		for i := 0; i < r.threadCount; i++ {
+			go func() {
+				defer wg.Done()
+				r.walk()
+			}()
+			// wait 1/2 second before starting a new thread
+			// TODO: why we are sleeping here?
```
Comment (FZambia) [inline pkg/services/thumbs/crawler.go:159]: ?

File: pkg/services/thumbs/models.go [hunk 7]
```diff
@@ -52,17 +52,17 @@ type crawlStatus struct {
...
-	// Assumes you have already authenticated as admin
+	// Stop assumes you have already authenticated as admin.
 	Stop() (crawlStatus, error)
 
-	// Assumes you have already authenticated as admin
+	// Status assumes you have already authenticated as admin.
 	Status() (crawlStatus, error)
 }
 
+// TODO: pass context to repo methods.
```
Comment (FZambia) [inline pkg/services/thumbs/models.go:65]: ?

File: pkg/services/thumbs/service.go [hunk 8]
```diff
@@ -47,15 +43,18 @@ func ProvideService(cfg *setting.Cfg, features featuremgmt.FeatureToggles, rende
...
 }
 
 type thumbService struct {
 	renderer      dashRenderer
 	thumbnailRepo thumbnailRepo
+	logger        log.Logger
 }
 
 func (hs *thumbService) Enabled() bool {
+	// TODO: Feature toggle check here?
```
Comment (FZambia) [inline pkg/services/thumbs/service.go:57]: ?

File: pkg/services/thumbs/crawler.go (hunk 3 above)
Comment (ArturWierzbicki) [reply pkg/services/thumbs/crawler.go:167]: oh thats right, didn't notice it - removed

File: pkg/services/thumbs/service.go (hunk 8 above)
Comment (ArturWierzbicki) [reply pkg/services/thumbs/service.go:57]: done https://github.com/grafana/grafana/pull/44891/files#diff-ebae43dace00a28a51fa647b1af7852dd443af155c2f46f7b2a089b1fc9736eaR88

File: pkg/services/thumbs/crawler.go (hunk 6 above)
Comment (ArturWierzbicki) [reply pkg/services/thumbs/crawler.go:159]: removed

File: pkg/services/thumbs/crawler.go (hunk 5 above)
Comment (ArturWierzbicki) [reply pkg/services/thumbs/crawler.go:122]: added early exit

**Pull request closed without merging**